uaclient/tests/test_apt.py

```diff
@@ -347,140 +347,190 @@ def test_errors_on_apt_helper_process_timeout(
 
 
 class TestAddAuthAptRepo:
+    @pytest.mark.parametrize("series", ("xenial", "noble"))
     @mock.patch("uaclient.apt.gpg.export_gpg_key")
     @mock.patch("uaclient.system.subp")
     @mock.patch("uaclient.apt.get_apt_auth_file_from_apt_config")
     @mock.patch("uaclient.apt.assert_valid_apt_credentials")
-    @mock.patch(
-        "uaclient.system.get_release_info",
-        return_value=mock.MagicMock(series="xenial"),
-    )
+    @mock.patch("uaclient.system.get_release_info")
     def test_add_auth_apt_repo_writes_sources_file(
         self,
         m_get_release_info,
         m_valid_creds,
         m_get_apt_auth_file,
         m_subp,
         m_gpg_export,
+        series,
         tmpdir,
     ):
         """Write a properly configured sources file to repo_filename."""
         repo_file = tmpdir.join("repo.conf").strpath
         auth_file = tmpdir.join("auth.conf").strpath
         m_get_apt_auth_file.return_value = auth_file
         m_subp.return_value = "500 esm.canonical.com...", ""  # apt policy
+        m_get_release_info.return_value = mock.MagicMock(series=series)
 
         add_auth_apt_repo(
             repo_filename=repo_file,
             repo_url="http://fakerepo",
             credentials="mycreds",
-            suites=("xenial",),
+            suites=(series,),
             keyring_file="keyring",
         )
 
-        expected_content = (
-            "deb http://fakerepo xenial main\n"
-            "# deb-src http://fakerepo xenial main\n"
-        )
+        if series == "xenial":
```

Comment:
I know the test is being run with "xenial" and "noble" as series, but perhaps here to use less hardcoding, what do you think about checking `if series in apt.SERIES_NOT_USING_DEB822` instead?

Reply:
Alternatively, we can parametrize the `expected_content` and `gpg_export_calls` alongside the `series`, so we won't need an if statement here at all.

Reply:
I think parametrizing everything makes sense, but @panlinux 's suggestion brings better readability - done that, let me know if this suffices